File: app/src/main/java/igrek/todotree/intent/GUICommand.java
package igrek.todotree.intent;


import javax.inject.Inject;

import igrek.todotree.app.AppData;
import igrek.todotree.app.AppState;
import igrek.todotree.dagger.DaggerIoc;
import igrek.todotree.domain.treeitem.AbstractTreeItem;
import igrek.todotree.service.tree.TreeManager;
import igrek.todotree.service.tree.TreeScrollCache;
import igrek.todotree.service.tree.TreeSelectionManager;
import igrek.todotree.ui.GUI;

public class GUICommand {
	
	@Inject
	TreeManager treeManager;
	
	@Inject
	GUI gui;
	
	@Inject
	AppData appData;
	
	@Inject
	TreeScrollCache scrollCache;
	
	@Inject
	TreeSelectionManager selectionManager;
	
	public GUICommand() {
		DaggerIoc.factoryComponent.inject(this);
	}
	
	public void updateItemsList() {
		appData.setState(AppState.ITEMS_LIST);
		gui.updateItemsList(treeManager.getCurrentItem(), null, selectionManager.getSelectedItems());
	}
	
	public void showItemsList() {
		appData.setState(AppState.ITEMS_LIST);
		gui.showItemsList(treeManager.getCurrentItem());
	}
	
	public void restoreScrollPosition(AbstractTreeItem parent) {
		Integer savedScrollPos = scrollCache.restoreScrollPosition(parent);
		if (savedScrollPos != null) {
			gui.scrollToPosition(savedScrollPos);
		}
	}
	
	public void guiInit() {
		gui.lazyInit();
	}
	
	public void numKeyboardHyphenTyped() {
		gui.quickInsertRange();
	}
}

File: app/src/main/java/igrek/todotree/intent/GUICommand.kt
package igrek.todotree.intent

import igrek.todotree.app.AppData
import igrek.todotree.app.AppState
import igrek.todotree.inject.LazyExtractor
import igrek.todotree.inject.LazyInject
import igrek.todotree.inject.appFactory
import igrek.todotree.service.tree.TreeManager
import igrek.todotree.ui.GUI
import igrek.todotree.ui.treelist.TreeListLayout

class GUICommand(
    treeManager: LazyInject<TreeManager> = appFactory.treeManager,
    gui: LazyInject<GUI> = appFactory.gui,
    appData: LazyInject<AppData> = appFactory.appData,
) {
    private val treeManager by LazyExtractor(treeManager)
    private val gui by LazyExtractor(gui)
    private val appData by LazyExtractor(appData)
    private val treeListLayout: TreeListLayout by LazyExtractor(appFactory.treeListLayout)

    fun updateItemsList() {
        appData.state = AppState.ITEMS_LIST
        treeListLayout.updateItemsList()
    }

    fun updateOneListItem(position: Int) {
        treeListLayout.updateOneListItem(position)
    }

    fun showItemsList() {
        appData.state = AppState.ITEMS_LIST
        treeManager.currentItem?.let { _ ->
            gui.showItemsList()
        }
    }

}